utils/handles/signal.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include "utils/handles/signal.h"
+
+#if !defined(_WIN32) && defined(__unix__) || defined(__unix) || \
+    (defined(__APPLE__) && defined(__MACH__))
+
+/*
+ * @para universal_signal_num_t SIG: assign the signal number which is going to
+ * handle.
+ * @para void (*signal_handler)(): signal handler of each signal needs
+ */
+__sighandler_t register_signal(universal_signal_num_t SIG,
```

Comment:
Since `register_signal` is just a wrapper of OS-specific functions, you can move it to header `utils/handles/signal.h` in the form of `static inline` functions which can be optimized by compilers.